From 3d6a1405f873d6f4498fef77e2c9d8092a2bff3d Mon Sep 17 00:00:00 2001
From: Rastrisr <rastrisr@tutamail.com>
Date: Thu, 15 Oct 2026 05:37:21 +0000
Subject: [PATCH] NetworkStack: non-Google default captive portal URLs

AXP fork: default captive portal / connectivity check to captiveportal.kuketz.de (public, privacy-respecting; selectable in AXP Settings toggle) instead of Google.

Change-Id: I7252710224952c005bc70053a18732fbd2db4c57
---
 res/values/config.xml                                    | 4 ++--
 src/com/android/networkstack/util/NetworkStackUtils.java | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/res/values/config.xml b/res/values/config.xml
index 8d6b64e..9f60884 100644
--- a/res/values/config.xml
+++ b/res/values/config.xml
@@ -13,7 +13,7 @@
          config_captive_portal_http_url and *NOT* by changing or overlaying
          this resource. It will break if the enforcement of overlayable starts.
          -->
-    <string name="default_captive_portal_http_url" translatable="false">http://connectivitycheck.gstatic.com/generate_204</string>
+    <string name="default_captive_portal_http_url" translatable="false">http://captiveportal.kuketz.de</string>
     <!-- HTTPS URL for network validation, to use for confirming internet connectivity. -->
     <!-- default_captive_portal_https_url is not configured as overlayable so
          OEMs that wish to change captive_portal_https_url configuration must
@@ -21,7 +21,7 @@
          config_captive_portal_https_url and *NOT* by changing or overlaying
          this resource. It will break if the enforcement of overlayable starts.
          -->
-    <string name="default_captive_portal_https_url" translatable="false">https://www.google.com/generate_204</string>
+    <string name="default_captive_portal_https_url" translatable="false">https://captiveportal.kuketz.de</string>
 
     <!-- List of fallback URLs to use for detecting captive portals. -->
     <!-- default_captive_portal_fallback_urls is not configured as overlayable
diff --git a/src/com/android/networkstack/util/NetworkStackUtils.java b/src/com/android/networkstack/util/NetworkStackUtils.java
index fce06e4..ac1a977 100755
--- a/src/com/android/networkstack/util/NetworkStackUtils.java
+++ b/src/com/android/networkstack/util/NetworkStackUtils.java
@@ -141,14 +141,14 @@ public class NetworkStackUtils {
      * The default list of HTTP URLs to use for detecting captive portals.
      */
     public static final String[] DEFAULT_CAPTIVE_PORTAL_HTTP_URLS =
-            new String [] {"http://connectivitycheck.gstatic.com/generate_204"};
+            new String [] {"http://captiveportal.kuketz.de"};
 
     /**
      * The default list of HTTPS URLs for network validation, to use for confirming internet
      * connectivity.
      */
     public static final String[] DEFAULT_CAPTIVE_PORTAL_HTTPS_URLS =
-            new String [] {"https://www.google.com/generate_204"};
+            new String [] {"https://captiveportal.kuketz.de"};
 
     /**
      * Minimum module version at which to enable the DHCP Rapid Commit option.
-- 
2.43.0

